### PR TITLE
[core] Remove v4 conditional code

### PR DESCRIPTION
**Pull request by flaviendelangle**
I wait for #2571 before doing those who are more than just remove some branches (for instance replace the `muiStyleAlpha` function by a direct call to `alpha` from the core

Preview: https://deploy-preview-2575--material-ui-x.netlify.app/components/data-grid/

### CODE REVIEW

File: packages/grid/_modules_/grid/utils/utils.ts [hunk 1]
```diff
@@ -15,23 +15,6 @@ export function isObject(value: any): value is Record<string, any> {
...
-  }
-
-  if ('fade' in styles && 'alpha' in styles) {
-    return 'v4.12';
-  }
-
-  return 'v4';
-}
-
 export function muiStyleAlpha(color: string, value: number): string {
```
Comment (m4theushw) [inline packages/grid/_modules_/grid/utils/utils.ts:18]: This helper can be removed now.

File: packages/grid/_modules_/grid/components/GridPagination.tsx [hunk 2]
```diff
@@ -60,26 +60,12 @@ export const GridPagination = React.forwardRef<
   );
 
   const handlePageChange = React.useCallback(
```
Comment (m4theushw) [inline packages/grid/_modules_/grid/components/GridPagination.tsx:62]:
```suggestion
  const handlePageChange = React.useCallback<TablePaginationProps['onPageChange']>(
```

File: packages/grid/_modules_/grid/components/containers/GridRootStyles.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+import { muiStyleAlpha, createTheme } from '../../utils/utils';
 import { gridClasses } from '../../gridClasses';
 
 const defaultTheme = createTheme();
 export const useStyles = makeStyles(
   (theme: Theme) => {
     const borderColor =
-      getThemePaletteMode(theme.palette) === 'light'
+      theme.palette.mode === 'light'
         ? lighten(muiStyleAlpha(theme.palette.divider, 1), 0.88)
```
Comment (m4theushw) [inline packages/grid/_modules_/grid/components/containers/GridRootStyles.ts:11]: `alpha` from `@mui/material/styles` can be used now.

```suggestion
        ? lighten(alpha(theme.palette.divider, 1), 0.88)
```

File: packages/grid/_modules_/grid/utils/utils.ts (hunk 1 above)
Comment (flaviendelangle) [reply packages/grid/_modules_/grid/utils/utils.ts:18]: Yes some changes needed you migration to `@mui/`, I'll complet this PR :+1:

File: packages/grid/_modules_/grid/components/GridPagination.tsx (hunk 2 above)
Comment (flaviendelangle) [reply packages/grid/_modules_/grid/components/GridPagination.tsx:61]: A lot better indeed

Comment (m4theushw) [reply packages/grid/_modules_/grid/components/GridPagination.tsx]: It won't work with `onRowsPerPageChange`, I tried. Maybe the typings in the core are wrong.

**Review by oliviertassinari — COMMENTED** [review]
It looks like it's going in the right direction.

**Comment by flaviendelangle** [comment]
@oliviertassinari it's always a pleasure to remove this kind of temporary code

**Comment by flaviendelangle** [comment]
Fixing this comment https://github.com/mui-org/material-ui-x/blob/next/packages/grid/_modules_/grid/components/columnSelection/GridHeaderCheckbox.tsx#L51 causes the related test to fail

I'll check and remove the comment if the v5 still needs the `|| isIndeterminate`

**Comment by flaviendelangle** [comment]
I think we can move on without https://github.com/mui-org/material-ui-x/pull/2575#issuecomment-919251610 and fix the more complex one later

**Pull request merged**